backend/internal/endpointctx/endpointctx.go

```diff
@@ -0,0 +1,5 @@
+package internal
+
+type EndpointCtxKeyType struct{}
```

Comment:
a comment here explaining this weird package may be useful in the future

Reply:
Added comment!